src/main/drivers/gyro_sync.c

```diff
@@ -23,24 +22,35 @@ bool gyroSyncCheckUpdate(gyroDev_t *gyro)
     return gyro->intStatus(gyro);
 }
 
-uint32_t gyroSetSampleRate(uint8_t lpf, uint8_t gyroSyncDenominator)
+uint32_t gyroSetSampleRate(gyroDev_t *gyro, uint8_t lpf, uint8_t gyroSyncDenominator, bool gyro_use_32khz)
 {
+#ifndef GYRO_SUPPORTS_32KHZ
+    if (gyro_use_32khz) {
+        gyro_use_32khz = false;
+    }
+#endif
     int gyroSamplePeriod;
 
     if (lpf == GYRO_LPF_256HZ || lpf == GYRO_LPF_NONE) {
+        gyro->gyroRateKHz = GYRO_RATE_8_kHz;
         gyroSamplePeriod = 125;
```

Comment:
also the sampleperiod is actually 31,5 I believe? maybe sampleperiod should be float?